src/components/HomePage/RepositoryList/index.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { useMemo } from "react";
+import { RepositoryListProps } from "../../../interfaces";
+import "./RepositoryList.css";
+import RepositoryListItem from "./RepositoryListItem";
+import SortDropdown from "./SortDropdown";
+import { RootState } from "../../../redux/rootReducer";
+import { useSelector } from "react-redux";
+
+const getFavoritesData = (state: RootState) => state.favorites;
+
+const RepositoryList = ({ repository }: RepositoryListProps) => {
+  const favoritesData = useSelector(getFavoritesData);
+  const favIDs = useMemo(
+    () => favoritesData.favorites.map((fav) => fav.id),
+    [favoritesData.favorites]
+  );
+
+  const isFavoriteData = useMemo(
+    () => repository.map((repository) => favIDs.includes(repository.id)),
+    [favIDs, repository]
+  );
```

Comment:
Suppose there are 10k Repos loaded because of pagination, this will result in unnecessary space utilization. Better is to keep a favorite map, and then while render you can do `favouritesMap[id]`

Reply:
https://github.com/himanshu-spr/github-fav-react/pull/2#discussion_r685110990
https://github.com/himanshu-spr/github-fav-react/pull/2#discussion_r685111058

These issues been already fixed

Reply:
https://github.com/himanshu-spr/github-fav-react/pull/2#discussion_r685111058
https://github.com/himanshu-spr/github-fav-react/pull/2#discussion_r685110990

These issues been already fixed